usagov/usagov-2021
# Usagov 1793 reduce unused css

## Jira Task


https://cm-jira.usa.gov/browse/USAGOV-USAGOV1793

## Description

Instead of forwarding all uswds packages, only forward the necessary ones in styles.scss in order to build a smaller styles.css file

forwarded packages: 
@forward "usa-accordion";
@forward "usa-alert";
@forward "usa-banner";
@forward "usa-breadcrumb";
@forward "usa-button";
@forward "usa-button-group";
@forward "usa-card";
@forward "usa-checkbox";
@forward "usa-combo-box";
@forward "usa-footer";
@forward "usa-header";
@forward "usa-hero";
@forward "usa-icon";
@forward "usa-icon-list";
@forward "usa-identifier";
@forward "usa-input";
@forward "usa-nav";
@forward "usa-process-list";
@forward "usa-radio";
@forward "usa-search";
@forward "usa-sidenav";
@forward "usa-table";
@forward "uswds-global";
@forward "uswds-typography";
@forward "uswds-utilities";

removed packages:
// @forward "usa-character-count";
// @forward "usa-checklist";
// @forward "usa-collection";
// @forward "usa-content";
// @forward "usa-dark-background";
// @forward "usa-date-picker";
// @forward "usa-date-range-picker";
// @forward "usa-display";
// @forward "usa-embed-container";
// @forward "usa-error-message";
// @forward "usa-fieldset";
// @forward "usa-file-input";
// @forward "usa-form";
// @forward "usa-form-group";
// @forward "usa-graphic-list";
// @forward "usa-hint";
// @forward "usa-input-prefix-suffix";
// @forward "usa-intro";
// @forward "usa-label";
// @forward "usa-layout-docs";
// @forward "usa-layout-grid";
// @forward "usa-legend";
// @forward "usa-link";
// @forward "usa-list";
// @forward "usa-media-block";
// @forward "usa-memorable-date";
// @forward "usa-modal";
// @forward "usa-pagination";
// @forward "usa-paragraph";
// @forward "usa-prose";
// @forward "usa-range";
// @forward "usa-section";
// @forward "usa-select";
// @forward "usa-site-alert";
// @forward "usa-skipnav";
// @forward "usa-step-indicator";
// @forward "usa-summary-box";
// @forward "uswds-form-controls";
// @forward "uswds-helpers";
// @forward "uswds-fonts";
// @forward "uswds-core";


## Type of Changes

  - Add screenshot showing what it should look like
- [x ] Other --> recompile styles.css

## Testing Instructions



cd into web/themes/custum/usagov/ and run npx gulp compile

cd .. until back in usagov-2021

run bin/drush cr, and make sure localhost styling is the same as prod. Helpful to use this document to double check the design components, going through the examples of differen uswds on the usagov site and making sure they look correct. https://docs.google.com/document/d/1gnWMeSAM-_-X4bC_K2nbyUMXugMNJlR9r_yFKmLfFSw/edit go to web standards section


### Change Requirements



### Validation Steps

- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review




## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.
